ZAPD/ZDisplayList.cpp

```diff
@@ -43,13 +42,23 @@ void ZDisplayList::ExtractFromXML(tinyxml2::XMLElement* reader, uint32_t nRawDat
 {
 	rawDataIndex = nRawDataIndex;
 	ParseXML(reader);
+	//TODO add error handling here
+	if (Globals::Instance->game == ZGame::OOT_SW97 || (reader->Attribute("Ucode", "f3dex")))
+	{
+		dListType = DListType::F3DEX;
+	}
+	else
+	{
+		dListType = DListType::F3DZEX;
+	}
```

Comment:
As I told you yesterday, currently it is preferred to access the XML attribute values using  `registeredAttributes` instead of using the reader directly.
Judging by the code, you may be interested on the `.wasSet` attribute of the registered attributes system.

You should also add this new feature to the docs (`docs/zapd_extraction_xml_reference.md`).

Finally, this should either print a warning or give an error if an invalid value is provided by the user. There are a few examples on how to use the WarningHandler system, but if you need help don't doubt in asking.